lib/rules/prefer-template.js

```diff
@@ -187,6 +201,14 @@ module.exports = {
                 return sourceCode.getText(currentNode);
             }
 
+            if (isConcatenation(currentNode) && !hasIdentifierReference(currentNode)) {
```

Comment:
This seems to be working well when the current node represents a subtree of multiple string literals at the start of the whole chain, but produces a wrong output if those subtrees appear in other positions.

For example:

```js
"a" + foo + ("c" + "d");
```

gets autofixed to:

```js
`a${  foo  }c" + `d";
```

Reply:
Updated and added more test cases 👍🏻